File: backend-dotnet/Application/Commands/Comments/DeleteCommentCommandHandler.cs
using Application.Interfaces;
using Domain.Entities;

namespace Application.Commands.Comments
{
    public class DeleteCommentCommandHandler : ICommandHandler<DeleteCommentCommand>
    {
        private readonly ICommentRepository _commentRepository;
        private readonly INotificationRepository _notificationRepository;

        public DeleteCommentCommandHandler(
            ICommentRepository commentRepository,
            INotificationRepository notificationRepository)
        {
            _commentRepository = commentRepository;
            _notificationRepository = notificationRepository;
        }

        public async Task<bool> HandleAsync(DeleteCommentCommand command)
        {
            var comment = await _commentRepository.GetByIdAsync(command.CommentId);
            if (comment == null)
                throw new KeyNotFoundException("Comment not found.");

            if (comment.UserId != command.UserId)
                throw new UnauthorizedAccessException("You are not the owner of this comment.");

            _commentRepository.Remove(comment);

            var notification = await _notificationRepository.GetCommentNotificationAsync(comment.PostId, command.UserId);
            if (notification != null)
                _notificationRepository.Remove(notification);

            await _commentRepository.SaveChangesAsync();
            return true;
        }
    }
}

File: backend-java/src/main/java/com/jerrygram/application/commands/comments/DeleteCommentCommandHandler.java
package com.jerrygram.application.commands.comments;

import com.jerrygram.application.interfaces.ICommandHandler;
import com.jerrygram.infrastructure.repositories.CommentRepository;
import com.jerrygram.infrastructure.repositories.NotificationRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.NoSuchElementException;

@Service
@RequiredArgsConstructor
@Slf4j
public class DeleteCommentCommandHandler implements ICommandHandler<DeleteCommentCommand, Boolean> {

    private final CommentRepository commentRepository;
    private final NotificationRepository notificationRepository;

    @Override
    @Transactional
    public Boolean handle(DeleteCommentCommand command) {
        var commentId = command.getCommentId();
        var userId = command.getUserId();
        
        log.info("Deleting comment: {} by user: {}", commentId, userId);

        var comment = commentRepository.findById(commentId)
                .orElseThrow(() -> new NoSuchElementException("Comment not found."));

        // Authorization check - only comment author can delete
        if (!comment.getUserId().equals(userId)) {
            throw new SecurityException("You are not the owner of this comment.");
        }
        
        // Delete comment
        commentRepository.delete(comment);
        
        // Delete related notification if exists
        try {
            notificationRepository.deleteCommentNotification(comment.getPostId(), userId);
            log.debug("Deleted comment notification for post {} by user {}", comment.getPostId(), userId);
        } catch (Exception e) {
            log.warn("Failed to delete comment notification: {}", e.getMessage());
        }
        
        log.info("Comment {} deleted successfully", commentId);
        return true;
    }
}